frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -240,14 +244,33 @@ const deleteWebhooks = (
 export const cleanUpWorkload = async (workload: OdcNodeModel): Promise<K8sResourceKind[]> => {
   const { resource } = workload;
   const reqs = [];
-  const isBuildConfigPresent = !_.isEmpty(workload.data?.resources?.buildConfigs);
   const isImageStreamPresent = await k8sGet(
     ImageStreamModel,
     resource.metadata.name,
     resource.metadata.namespace,
   )
     .then(() => true)
     .catch(() => false);
+  const buildConfigs = await k8sList(BuildConfigModel, { ns: resource.metadata.namespace });
+  const builds = await k8sList(BuildModel, { ns: resource.metadata.namespace });
+  const resources = {
+    buildConfigs: {
+      data: buildConfigs,
+      loaded: true,
+      loadError: null,
+    },
+    builds: {
+      data: builds,
+      loaded: true,
+      loadError: null,
+    },
+  };
+  const resourceBuildConfigs =
+    resource.kind === 'CronJob'
+      ? getBuildConfigsForCronJob(resource as CronJobKind, resources)
+      : getBuildConfigsForResource(resource, resources);
```

Comment:
Just an idea: We can move this check also into `getBuildConfigsForResource` ? So that it starts with

```tsx
export const getBuildConfigsForResource = (
  resource: K8sResourceKind,
  resources: any,
): BuildConfigOverviewItem[] => {
  if (resource.kind === 'CronJob') {
    return getBuildConfigsForCronJob(resource as CronJobKind, resources);
  }
  // ....
```

Makes this sense? Not required to get a lgtm :)